graph_weather/models/atmorep/model/field_transformer.py

```diff
@@ -0,0 +1,143 @@
+from typing import List, Optional
+
+import torch
+import torch.nn as nn
+from einops import rearrange
+
+
+class FieldVisionTransformer(nn.Module):
+    """
+    A Vision Transformer (ViT) designed to process individual fields in a model.
+    It extracts features from spatial patches across multiple time steps using
+    a transformer architecture.
+
+    Args:
+        hidden_dim (int): Dimensionality of the transformer embeddings.
+        patch_size (int): Size of each spatial patch (square).
+        spatial_dims (tuple[int, int]): Full spatial dimensions (height, width).
+        time_steps (int): Number of temporal steps in the input.
+        num_layers (int): Number of transformer blocks.
+        field_name (str): Name of the field this transformer is processing (e.g., 'temperature').
+        transformer_block_cls (nn.Module): Class to instantiate for each transformer block.
+            Defaults to a placeholder that you should replace with your actual TransformerBlock.
+    """
+
+    def __init__(
+        self,
+        hidden_dim: int,
+        patch_size: int,
+        spatial_dims: tuple[int, int],
+        time_steps: int,
+        num_layers: int,
+        field_name: str = "unknown_field",
+        transformer_block_cls=nn.Module,  # replace with your actual block class
```

Comment:
Same as for the decoder, its used in quite a specific way, so removing this as an option makes more sense.
```suggestion
```